errai-ioc/src/main/java/org/jboss/errai/ioc/rebind/ioc/bootstrapper/IOCProcessor.java

```diff
@@ -1315,21 +1315,24 @@ private boolean isAlternativeEnabled(final MetaClass type) {
 
   private boolean isEnabledByProperty(final MetaClass type) {
     final EnabledByProperty anno = type.getAnnotation(EnabledByProperty.class);
-    final boolean propValue = getPropertyValue(anno.value());
+    final boolean propValue = getPropertyValue(anno.value(),
+                                               anno.matchValue(),
+                                               anno.matchByDefault(),
+                                               anno.caseSensitive());
     final boolean negated = anno.negated();
 
     return propValue ^ negated;
   }
 
-  private boolean getPropertyValue(final String propName) {
+  protected boolean getPropertyValue(final String propName,
+                                     final String matchValue,
+                                     final boolean matchByDefault,
+                                     final boolean caseSensitive) {
     final String propertyValue = EnvUtil.getEnvironmentConfig().getFrameworkOrSystemProperty(propName);
-    if (propertyValue == null) {
-      return false;
-    }
-    try {
-      return Boolean.parseBoolean(propertyValue);
-    } catch (final Throwable t) {
-      throw new RuntimeException("Could not parse " + propName + " value, " + propertyValue + ", to boolean.", t);
+    if(propertyValue == null){
```

Comment:
Minor nit-pick, but could you please fix the formatting on this line so that there is space around the parentheses?